core/src/main/scala/filodb.core/memstore/PartKeyIndex.scala

```diff
@@ -69,14 +70,26 @@ abstract class PartKeyIndexRaw(ref: DatasetRef,
     .withTag("dataset", ref.dataset)
     .withTag("shard", shardNum)
 
+  protected val labelValuesQueryLatency = Kamon.histogram("index-label-values-query-latency",
+      MeasurementUnit.time.nanoseconds)
+    .withTag("dataset", ref.dataset)
+    .withTag("shard", shardNum)
+
   protected val queryIndexLookupLatency = Kamon.histogram("index-partition-lookup-latency",
       MeasurementUnit.time.nanoseconds)
     .withTag("dataset", ref.dataset)
     .withTag("shard", shardNum)
 
-  protected val indexDiskLocation = diskLocation.map(new File(_, ref.dataset + File.separator + shardNum))
+  protected val partIdFromPartKeyLookupLatency = Kamon.histogram("index-ingestion-partId-lookup-latency",
+      MeasurementUnit.time.nanoseconds)
+    .withTag("dataset", ref.dataset)
+    .withTag("shard", shardNum)
+
+  private val _indexDiskLocation = diskLocation.map(new File(_, ref.dataset + File.separator + shardNum))
     .getOrElse(createTempDir(ref, shardNum)).toPath
 
+  def indexDiskLocation: Path = _indexDiskLocation
```

Comment:
I need to save it to a member val because `createTempDir` being called every time would create a new folder on each ask of the location.

I have a wrapping method so that code like the test code can access this without going to the variable directly.

Reply:
Ah -- understood